scenario_player/node_support.py

```diff
@@ -68,109 +56,16 @@
-            archive = tar_archive
-        else:
-            zip_opener = ZipFile(release_file_path, "r")
-            with zip_opener as zip_archive:
-                contents = zip_archive.namelist()
-            archive = zip_archive
-
-        if len(contents) != 1:
-            raise ValueError(
-                f"Release archive has unexpected content. "
-                f'Expected 1 file, found {len(contents)}: {", ".join(contents)}'
-            )
-
-        bin_file_path = self._bin_path.joinpath(contents[0])
-        if not bin_file_path.exists():
-            log.debug(
-                "Extracting Raiden binary",
-                release_file_name=release_file_path.name,
-                bin_file_name=bin_file_path.name,
-            )
-            archive.extract(contents[0], str(self._bin_path))
-            bin_file_path.chmod(0o770)
-        return bin_file_path
-
-    def _get_release_file(self, release_file_name: str):
-        release_file_path = self._downloads_path.joinpath(release_file_name)
-        if release_file_path.exists():
-            return release_file_path
-
-        url = RAIDEN_RELEASES_URL + release_file_name
-        release_file_path.parent.mkdir(exist_ok=True, parents=True)
-        with requests.get(url, stream=True) as resp, release_file_path.open("wb+") as release_file:
-            log.debug("Downloading Raiden release", release_file_name=release_file_name)
-            if not 199 < resp.status_code < 300:
-                raise ValueError(
-                    f"Can't download release file {release_file_name}: "
-                    f"{resp.status_code} {resp.text}"
-                )
-            shutil.copyfileobj(resp.raw, release_file)
-        return release_file_path
-
-    @property  # type: ignore
-    @ttl_cache(maxsize=1, ttl=600)
-    def _latest_release_name(self):
-        latest_release_file_name = self._expand_release_template(
-            RAIDEN_RELEASES_LATEST_FILE_TEMPLATE
-        )
-        url = urljoin(RAIDEN_RELEASES_URL, latest_release_file_name)
-        log.debug("Fetching latest Raiden release", lookup_url=url)
-        return requests.get(url).text.strip()
-
-    @staticmethod
-    def _expand_release_template(template, **kwargs):
-        return template.format(
-            platform="macOS" if sys.platform == "darwin" else sys.platform,
-            arch=platform.machine(),
-            **kwargs,
-        )
-
-
 class NodeRunner:
-    def __init__(self, runner: "ScenarioRunner", index: int, raiden_version, options: dict):
+    def __init__(self, runner: "ScenarioRunner", index: int, raiden_client: str, options: dict):
         self._runner = runner
         self._index = index
-        self._raiden_version = raiden_version
         self._options = options
         self._nursery: Optional[Nursery] = None
+
+        # Overwrite raiden client to use, if given in options
+        self._raiden_client = options.pop("raiden-client", raiden_client)
```

Comment:
Just to assure, is option["raiden-client"] used somewhere afterwards? As it pops it from the dict.

Reply:
No, it isn't